clang/lib/CodeGen/TargetBuiltins/AMDGPU.cpp

```diff
@@ -585,6 +597,23 @@ Value *CodeGenFunction::EmitAMDGPUBuiltinExpr(unsigned BuiltinID,
     llvm::Value *Env = EmitScalarExpr(E->getArg(0));
     return Builder.CreateCall(F, {Env});
   }
+  case AMDGPU::BI__builtin_amdgcn_processor_is: {
+    assert(CGM.getTriple().isSPIRV() &&
```

Comment:
maybe check getTarget().getTargetOpts().CPU empty or generic instead? This could also be used by amdgcn triple with an empty target-cpu to generate some generic bitcode.

Reply:
I'm not entirely convinced that it's ideal to have multiple ways of generating "generic" IRs, one of which is mostly ad-hoc and only known to us, hence I'm not entirely convinced we want to encourage this. Yes, I appreciate that libc or ROCDL do this, but perhaps it's better to eventually off-ramp these into AMDGCN SPIR-V. If you two do not mind, I'd consider leaving this as a follow-up action, rather than doing it now.

Reply:
I am OK with leaving this for future